src/test/java/ArrayIndexedBagTest.java

```diff
@@ -0,0 +1,258 @@
+        bag.add(99);
+        assertFalse(bag.contains(101));
+    }
+
+    @Test
+    void getCountOfElementWithMultipleCopiesInIndexedBagReturnsCorrectCount() {
+        IndexedBag<Integer> bag = new ArrayIndexedBag<>();
+        bag.add(0);
+        bag.add(99);
+        bag.add(1);
+        bag.add(99);
+        bag.add(2);
+        bag.add(99);
+        assertEquals(3, bag.getCount(99));
+    }
+
+    @Test
+    void getCountOfElementNotInIndexedBagReturnsZero() {
+        IndexedBag<Integer> bag = new ArrayIndexedBag<>();
+        bag.add(0);
+        bag.add(99);
+        bag.add(1);
+        bag.add(99);
+        bag.add(2);
+        bag.add(99);
+        assertEquals(0, bag.getCount(101));
+    }
+
+    @Test
+    void addingToEndOfIndexedBagPutsElementInFIFOOrder() {
+        IndexedBag<Integer> bag = new ArrayIndexedBag<>();
+        for (int i = 0; i < 6; ++i) {
+            bag.add(i);
+        }
+        for (int i = 0; i < 6; ++i) {
+            assertEquals(i, bag.get(i));
+        }
+    }
+
+    @Test
+    void addingToIndexZeroOfIndexedBagPutsElementInLIFOOrder() {
+        IndexedBag<Integer> bag = new ArrayIndexedBag<>();
+        for (int i = 0; i < 6; ++i) {
+            bag.add(0, i);
+        }
+        int index = 0;
+        for (int value = 5; value >= 0; --value) {
+            assertEquals(index, bag.get(value));
+            index++;
+        }
+    }
+
+    @Test
+    void addingToSpecifiedIndicesInIndexedBagResultsInSpecifiedOrder() {
+        IndexedBag<Integer> bag = new ArrayIndexedBag<>();
+        bag.add(4);
+        bag.add(0, 0);
+        bag.add(1, 2);
+        bag.add(3, 5);
+        bag.add(2, 3);
+        bag.add(1, 1);
+        for (int i = 0; i < 6; ++i) {
+            assertEquals(i, bag.get(i));
+        }
+    }
+
+    @Test
+    void addingBeyondCapacityOfIndexedBagCallsExpandCapacityToMakeRoomWhileMaintainingOrdering() {
+        IndexedBag<Integer> bag = new ArrayIndexedBag<>(5);
+        for (int i = 0; i < 10; ++i) {
+            bag.add(i);
+        }
+        for (int i = 0; i < 10; ++i) {
+            assertEquals(i, bag.get(i));
+        }
+    }
+
+    @Test
+    void addingToIndexNotInRangeOfIndexedBagThrowsException() {
+        IndexedBag<Integer> bag = new ArrayIndexedBag<>();
+        assertThrows(IndexOutOfBoundsException.class, () -> bag.add(99, 101));
+    }
+
+    @Test
+    void gettingFromAnIndexNotInRangeOfIndexedBagThrowsException() {
+        IndexedBag<Integer> bag = new ArrayIndexedBag<>();
+        assertThrows(IndexOutOfBoundsException.class, () -> bag.get(99));
+    }
+
+    @Test
+    void getAndIndexOfCorrespondOnValueAndIndexInIndexedBag() {
+        IndexedBag<Integer> bag = new ArrayIndexedBag<>();
+        bag.add(101);
+        bag.add(102);
+        bag.add(103);
+        bag.add(104);
+        bag.add(2, 99);
+        assertEquals(bag.get(bag.indexOf(99)), 99);
+        assertEquals(bag.indexOf(bag.get(2)), 2);
+    }
```

Comment:
`get` seems unrelated to the test here, and is already tested in several other tests, like `addingToSpecifiedIndicesInIndexedBagResultsInSpecifiedOrder`.